### PR TITLE
SNOW-1629635 fix connection pool allocation

### PR DESCRIPTION
**Pull request by sfc-gh-mhofman**
### Description
Fixed connection pool manager allocation problem
Removal of exceptions related to optional connection string parameters

### Checklist
- [x] Code compiles correctly
- [x] Code is formatted according to [Coding Conventions](../blob/master/CodingConventions.md)
- [ ] Created tests which fail without the change (if possible)
- [x] All tests passing (`dotnet test`)
- [x] Extended the README / documentation, if necessary
- [x] Provide JIRA issue id (if possible) or GitHub issue id in PR name

### CODE REVIEW

**Review by sfc-gh-dstempniak — APPROVED** [review]

File: Snowflake.Data/Core/Session/SFSessionProperty.cs [hunk 1]
```diff
@@ -215,7 +215,7 @@ internal static SFSessionProperties ParseConnectionString(string connectionStrin
                 }
                 catch (ArgumentException e)
```
Comment (sfc-gh-pbulawa) [inline Snowflake.Data/Core/Session/SFSessionProperty.cs:216]: Can you remove variable `e` as it generates warnings.

Comment (sfc-gh-mhofman) [reply Snowflake.Data/Core/Session/SFSessionProperty.cs]: Fixed

**Review by sfc-gh-pbulawa — APPROVED** [review]

File: Snowflake.Data/Client/SnowflakeDbConnectionPool.cs [hunk 2]
```diff
@@ -129,6 +130,9 @@ internal static void SetConnectionPoolVersion(ConnectionPoolType requestedPoolTy
         {
             lock (s_connectionManagerInstanceLock)
             {
+                if (s_connectionManager != null)
```
Comment (sfc-gh-knozderko) [inline Snowflake.Data/Client/SnowflakeDbConnectionPool.cs:133]: it would disable the possibility to switch to the old pool

**Review by sfc-gh-knozderko — CHANGES_REQUESTED** [review]

File: Snowflake.Data/Client/SnowflakeDbConnectionPool.cs (hunk 2 above)
Comment (sfc-gh-mhofman) [reply Snowflake.Data/Client/SnowflakeDbConnectionPool.cs]: Fixed

**Review by sfc-gh-knozderko — APPROVED** [review]

**Pull request merged**